frontend/packages/console-plugin-sdk/src/typings/features.ts

```diff
@@ -12,7 +12,7 @@ namespace ExtensionProperties {
 
   export interface ActionFeatureFlag {
     /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
-    detect: (dispatch: Dispatch) => Promise<any>;
+    detect: (dispatch: Dispatch) => Promise<any> | void;
```

Comment:
The `Promise` result should indicate that `detect` function is async in nature.

Is there a specific reason for `| void`  union?

Reply:
k8sWatch would not return anything as it uses websockets.